test/python/WMCore_t/MicroService_t/MicroService_t.py

```diff
@@ -80,28 +82,86 @@ def testGetStatus(self):
         api = "status"
         url = '%s/%s' % (self.url, api)
         params = {}
-        data = self.mgr.getdata(url, params=params, encode=True, decode=True)
-        self.assertEqual(data['result'][0]['microservice'], self.managerName)
-        self.assertEqual(data['result'][0]['api'], api)
-
-        params = {"service": "transferor"}
-        data = self.mgr.getdata(url, params=params, encode=True, decode=True)
-        self.assertEqual(data['result'][0]['microservice'], self.managerName)
-        self.assertEqual(data['result'][0]['api'], api)
+        # perform check with and without gzip encoding
+        for hdr in ['', 'gzip']:
+            headers = {'Accept-Encoding': hdr}
+            data = self.mgr.getdata(url, params=params, headers=headers, encode=True, decode=True)
+            # check if our data is returned as bytes (gzip encoding)
+            if isinstance(data, bytes) and hdr == 'gzip':
+                data = gzip.decompress(data)
+                if isinstance(data, bytes):
+                    data = data.decode("utf-8")
+                data = json.loads(data)
+            print("### testGetStatus, headers=", headers, "data=", data, type(data))
+            # check that result key is present in data
+            if 'result' in data:
+                self.assertEqual(data['result'][0]['microservice'], self.managerName)
+                self.assertEqual(data['result'][0]['api'], api)
+            else:
+                print("### result is not present in data")
+                print(data)
+                self.assertEqual('result' in data, True)
```

Comment:
I would suggest to change it to `self.assertIsTrue('result' in data)`. But honestly, this line doesn't make sense because if we are in the `else` block, well, then there shouldn't be a `result` key in data...

Reply:
you are right here, I think it is oversight from my side. `self.assertEqual` can be easily remove because of if/else block.

Reply:
and I removed the self.assertEqual too